### ISSUE TITLE
a11y - edit page - add block/blocks-chooser search field - cms ui 

### ISSUE THREAD

**Issue by Wagner3UB** [plone/volto#5214]
**Describe the bug**
After typing something in the "search" field inside the "block chooser" modal an "X" appears to cancel the typed text, but this button has an aria-label "search". Another problem is when the button is focused, it only changes color. It is insufficient for accessibility. The same occours with the input search field, doesn't have a proper focus.

- [x] Button label
- [x] Button focus
- [x] input search field focus

**To Reproduce**

1- Go to https://demo.plone.org/edit
2- enter in edit mode by clicking on the edit button (top-left)
3- Press enter to create a new content
4- Click on the "add block" icon
5- Type some text, the "X" will appear inside the input

**Expected behavior**
Give a coherent name/aria-label to the button
Give proper focus to the button; the minimum requirement is to have borders.
Give proper focus to the input search field

**Screenshots**
<img width="453" alt="Schermata 2023-09-21 alle 16 57 51" src="https://github.com/plone/volto/assets/60133113/94ba3335-567d-4b73-acc3-681c57c1b190">

**Comment by vikaskumar2703** [comment]
I would like to work on this issue

**Comment by rishuraj1** [comment]
Hey @Wagner3UB I really want to contribute, please assign me to this issue and can you give me an heads up. Thanks.

**Comment by stevepiercy** [comment]
@rishuraj1 please read and follow [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html).

**Comment by rishuraj1** [comment]
Hey @Wagner3UB  I identified the issue and was able to work on it. Please look into #5254  . Thanks
